packager/packager.cc

```diff
@@ -1411,17 +1476,64 @@ ast::ParsedFile rewritePackage(core::Context ctx, ast::ParsedFile file) {
     return file;
 }
 
+string_view levelToSigil(core::StrictLevel level) {
+    switch (level) {
+        case core::StrictLevel::None:
+            Exception::raise("Should never happen");
+        case core::StrictLevel::Internal:
+            Exception::raise("Should never happen");
+        case core::StrictLevel::Ignore:
+            return "ignore"sv;
+        case core::StrictLevel::False:
+            return "false"sv;
+        case core::StrictLevel::True:
+            return "true"sv;
+        case core::StrictLevel::Strict:
+            return "strict"sv;
+        case core::StrictLevel::Strong:
+            return "strong"sv;
+        case core::StrictLevel::Max:
+            Exception::raise("Should never happen");
+        case core::StrictLevel::Autogenerated:
+            Exception::raise("Should never happen");
+        case core::StrictLevel::Stdlib:
+            return "__STDLIB_INTERNAL"sv;
+    }
+}
+
 ast::ParsedFile rewritePackagedFile(core::Context ctx, ast::ParsedFile parsedFile) {
     auto &file = parsedFile.file.data(ctx);
     ENFORCE(file.sourceType != core::File::Type::Package);
     auto &pkg = ctx.state.packageDB().getPackageForFile(ctx, ctx.file);
     if (pkg.exists()) {
         auto &pkgImpl = PackageInfoImpl::from(pkg);
 
+        auto isTestFile = core::packages::PackageDB::isTestFile(ctx, file);
+        // check that the files are indeed as typed as they should be
+        if (!isTestFile && file.originalSigil < pkgImpl.minStrictLevel()) {
+            auto [begin, end] = core::File::fileStrictSigilLocation(file.source());
+            if (auto e = ctx.beginError(core::LocOffsets{static_cast<uint32_t>(begin), static_cast<uint32_t>(end)},
+                                        core::errors::Packager::TypedLevelViolation)) {
+                e.setHeader("File `{}` is marked as `{}`, but it belongs to package `{}` which stipulates a minimum "
+                            "typed level of `{}`",
+                            ctx.file.data(ctx).path(), levelToSigil(file.originalSigil), pkgImpl.name.toString(ctx),
+                            levelToSigil(pkgImpl.minStrictLevel()));
+            }
+        }
+        if (isTestFile && file.originalSigil < pkgImpl.minTestStrictLevel()) {
+            auto [begin, end] = core::File::fileStrictSigilLocation(file.source());
+            if (auto e = ctx.beginError(core::LocOffsets{static_cast<uint32_t>(begin), static_cast<uint32_t>(end)},
```

Comment:
If we returned a `core::LocOffsets` in `fileStrictSigilLocation`, we could encapsulate this cast and pass along the parse result unchanged.

Reply:
This has been done.